Description: Fixes compatibility with SQLAlchemy
 This is a backport of what has been done at:
  https://review.openstack.org/#/c/36302/
Author: Thomas Goirand <zigo@debian.org>
Origin: https://review.openstack.org/#/c/36302/
Bug-Debian: http://bugs.debian.org/715431
Last-Update: 2013-07-10

--- cinder-2013.1.2.orig/cinder/db/sqlalchemy/migrate_repo/versions/008_add_backup.py
+++ cinder-2013.1.2/cinder/db/sqlalchemy/migrate_repo/versions/008_add_backup.py
@@ -36,52 +36,17 @@ def upgrade(migrate_engine):
         Column('deleted', Boolean(create_constraint=True, name=None)),
         Column('id', String(36), primary_key=True, nullable=False),
         Column('volume_id', String(36), nullable=False),
-        Column('user_id', String(length=255, convert_unicode=False,
-                                 assert_unicode=None,
-                                 unicode_error=None,
-                                 _warn_on_bytestring=False)),
-        Column('project_id', String(length=255, convert_unicode=False,
-                                    assert_unicode=None,
-                                    unicode_error=None,
-                                    _warn_on_bytestring=False)),
-        Column('host', String(length=255, convert_unicode=False,
-                              assert_unicode=None,
-                              unicode_error=None,
-                              _warn_on_bytestring=False)),
-        Column('availability_zone', String(length=255,
-                                           convert_unicode=False,
-                                           assert_unicode=None,
-                                           unicode_error=None,
-                                           _warn_on_bytestring=False)),
-        Column('display_name', String(length=255, convert_unicode=False,
-                                      assert_unicode=None,
-                                      unicode_error=None,
-                                      _warn_on_bytestring=False)),
-        Column('display_description', String(length=255,
-                                             convert_unicode=False,
-                                             assert_unicode=None,
-                                             unicode_error=None,
-                                             _warn_on_bytestring=False)),
-        Column('container', String(length=255, convert_unicode=False,
-                                   assert_unicode=None,
-                                   unicode_error=None,
-                                   _warn_on_bytestring=False)),
-        Column('status', String(length=255, convert_unicode=False,
-                                assert_unicode=None,
-                                unicode_error=None,
-                                _warn_on_bytestring=False)),
-        Column('fail_reason', String(length=255, convert_unicode=False,
-                                     assert_unicode=None,
-                                     unicode_error=None,
-                                     _warn_on_bytestring=False)),
-        Column('service_metadata', String(length=255, convert_unicode=False,
-                                          assert_unicode=None,
-                                          unicode_error=None,
-                                          _warn_on_bytestring=False)),
-        Column('service', String(length=255, convert_unicode=False,
-                                 assert_unicode=None,
-                                 unicode_error=None,
-                                 _warn_on_bytestring=False)),
+        Column('user_id', String(length=255)),
+        Column('project_id', String(length=255)),
+        Column('host', String(length=255)),
+        Column('availability_zone', String(length=255)),
+        Column('display_name', String(length=255)),
+        Column('display_description', String(length=255)),
+        Column('container', String(length=255)),
+        Column('status', String(length=255)),
+        Column('fail_reason', String(length=255)),
+        Column('service_metadata', String(length=255)),
+        Column('service', String(length=255)),
         Column('size', Integer()),
         Column('object_count', Integer()),
         mysql_engine='InnoDB'
--- cinder-2013.1.2.orig/cinder/db/sqlalchemy/migrate_repo/versions/002_quota_class.py
+++ cinder-2013.1.2/cinder/db/sqlalchemy/migrate_repo/versions/002_quota_class.py
@@ -35,18 +35,10 @@ def upgrade(migrate_engine):
                                                     name=None)),
                           Column('id', Integer(), primary_key=True),
                           Column('class_name',
-                                 String(length=255,
-                                        convert_unicode=True,
-                                        assert_unicode=None,
-                                        unicode_error=None,
-                                        _warn_on_bytestring=False),
+                                 String(length=255),
                                  index=True),
                           Column('resource',
-                                 String(length=255,
-                                        convert_unicode=True,
-                                        assert_unicode=None,
-                                        unicode_error=None,
-                                        _warn_on_bytestring=False)),
+                                 String(length=255)),
                           Column('hard_limit', Integer(), nullable=True),
                           mysql_engine='InnoDB',
                           mysql_charset='utf8',
@@ -66,14 +58,9 @@ def upgrade(migrate_engine):
                                                    name=None)),
                          Column('id', Integer(), primary_key=True),
                          Column('project_id',
-                                String(length=255, convert_unicode=True,
-                                       assert_unicode=None, unicode_error=None,
-                                       _warn_on_bytestring=False),
-                                index=True),
+                                String(length=255),index=True),
                          Column('resource',
-                                String(length=255, convert_unicode=True,
-                                       assert_unicode=None, unicode_error=None,
-                                       _warn_on_bytestring=False)),
+                                String(length=255)),
                          Column('in_use', Integer(), nullable=False),
                          Column('reserved', Integer(), nullable=False),
                          Column('until_refresh', Integer(), nullable=True),
@@ -95,25 +82,17 @@ def upgrade(migrate_engine):
                                                    name=None)),
                          Column('id', Integer(), primary_key=True),
                          Column('uuid',
-                                String(length=36,
-                                       convert_unicode=True,
-                                       assert_unicode=None,
-                                       unicode_error=None,
-                                       _warn_on_bytestring=False),
+                                String(length=36),
                                 nullable=False),
                          Column('usage_id',
                                 Integer(),
                                 ForeignKey('quota_usages.id'),
                                 nullable=False),
                          Column('project_id',
-                                String(length=255, convert_unicode=True,
-                                       assert_unicode=None, unicode_error=None,
-                                       _warn_on_bytestring=False),
+                                String(length=255),
                                 index=True),
                          Column('resource',
-                                String(length=255, convert_unicode=True,
-                                       assert_unicode=None, unicode_error=None,
-                                       _warn_on_bytestring=False)),
+                                String(length=255)),
                          Column('delta', Integer(), nullable=False),
                          Column('expire', DateTime(timezone=False)),
                          mysql_engine='InnoDB',
